### PR TITLE
Add RGB raw mode for color sensor

### PR DESCRIPTION
**Pull request by THEb0nny**
We did extra mode. I checked it in a local makecode. It works for me.

### CODE REVIEW

File: libs/color-sensor/color.ts [hunk 1]
```diff
@@ -93,6 +93,10 @@ namespace sensors {
                 || this.mode == ColorSensorMode.AmbientLightIntensity
                 || this.mode == ColorSensorMode.ReflectedLightIntensity)
                 return this.getNumber(NumberFormat.UInt8LE, 0)
+            if (this.mode == ColorSensorMode.RgbRaw)
+                return this.getNumber(NumberFormat.UInt16LE, 0)
+            if (this.mode == ColorSensorMode.RefRaw)
```
Comment (THEb0nny) [inline libs/color-sensor/color.ts]: It is not necessary. This is for the next request

File: libs/color-sensor/color.ts [hunk 2]
```diff
@@ -118,6 +120,8 @@ namespace sensors {
                 control.raiseEvent(this._id, this._colorEventValue(curr));
             else
                 this.thresholdDetector.setLevel(curr);
+            if (this.mode == ColorSensorMode.RgbRaw)
```
Comment (abchatra) [inline libs/color-sensor/color.ts]: Is it ok to set threasholdDetector before raising the event?

File: libs/color-sensor/color.ts [hunk 3]
```diff
@@ -179,6 +183,28 @@ namespace sensors {
             return this.getNumber(NumberFormat.UInt8LE, 0)
         }
 
+        /**
+         * Get the current rgb values from the color sensor.
+         * @param sensor the color sensor to query the request
+         */
+        //% help=sensors/color-sensor/rgbraw
+        //% block="**color sensor** %this| rgbraw"
```
Comment (abchatra) [inline libs/color-sensor/color.ts]: In my opinon this is advanced and we don't add blocks for advanced APIs. Make this a JavaScript only api.

Comment (THEb0nny) [reply libs/color-sensor/color.ts]: Sorry, did not quite understand. I think that RGB is very necessary for solving problems of determining color in rgb and translating through this mode in hsv. For competition, we need an RGB mode.

**Comment by THEb0nny** [comment]
@pelikhan need your review..

**Comment by THEb0nny** [comment]
@abchatra you can ask for a review from moskal?

**Comment by abchatra** [comment]
This is an API addition and we need to go through API review. Give me some time to get it done. Sorry for the bit delay here. I will come back with the recommendation.

**Comment by THEb0nny** [comment]
I combined rgbraw and refraw into one pull.
https://github.com/microsoft/pxt-ev3/pull/876

**Pull request closed without merging**